### DISCUSSION TITLE
mx-auto does not center the element

### DISCUSSION THREAD

**Discussion by Chandler-Zhu** [tailwindlabs/tailwindcss#8775] [Help] [answered]
I wanted to center a button horizontally with `mx-auto` class but it didn't work
repro: https://play.tailwindcss.com/Cpp0uNf4yJ

**Comment by clovertwin** [comment, marked as answer]
Adding text-center to the parent divs will make it work.

**Reply by clovertwin** [reply]
Or add display block to the buttons :)

**Reply by Chandler-Zhu** [reply]
adding `display: block;` does the trick for me, but if I want to center the button vertically by using `my-auto` , it does not work
https://play.tailwindcss.com/1lGrX5XdTG

**Reply by clovertwin** [reply]
If both margin-top and margin-bottom are auto, the value used for both of them is zero. I would suggest using flex-box instead. If you remove `mx-auto my-auto block` from the button, and add `flex justify-center items-center` to the parent div it should work. Check out this article about it: https://stackoverflow.com/questions/12415661/using-marginauto-to-vertically-align-a-div
